miasm2/arch/aarch64/sem.py

```diff
@@ -124,31 +124,32 @@ def extend_arg(dst, arg):
         return arg
 
     op, (reg, shift) = arg.op, arg.args
-    if op == 'SXTW':
-        base = reg.signExtend(dst.size)
+    if op == "SXTB":
+        base = reg[:8].signExtend(dst.size)
         op = "<<"
-    elif op == "SXTB":
-        base = reg.signExtend(dst.size)
+    elif op == 'SXTW':
+        base = reg[:16].signExtend(dst.size)
         op = "<<"
     elif op == "SXTH":
-        base = reg.signExtend(dst.size)
+        base = reg[:32].signExtend(dst.size)
         op = "<<"
```

Comment:
Ok I did it again: My comment was wrong again: the `SXTW` is *word* sign extend so:
```python
     if op == 'SXTW':
         base = reg[:32].signExtend(dst.size)
         op = "<<"
```

The 16 bit is for `SXTH` !

So all the *W/*H are inverted. Sorry for that :confused:
By the way, can you squash your commits into one?

Reply:
Gotcha. Fixed it, hope got it right the 3rd time :).